kineticTypo/kineticTypo1/Tile.pde

```diff
@@ -61,36 +67,40 @@ class Tile {
   }
  
   void display(){
-    pushMatrix(); // 
+    pushMatrix(); //keeps coordinate system for each tile at 0,0 
     translate(this.posX,this.posY);
     this.drawAnimation();
     image(this.pg, 0, 0);
     popMatrix();
   }
+  
  
   void drawAnimation(){
+    
     pg.beginDraw();
     for (int fy = 0; fy < Tile.fraqY; fy++) { // wrap up 2D loop
       for (int fx = 0; fx < Tile.fraqX; fx++) {
         // WAVE
-        int wave = int(sin((frameCount + ( fx*fy )) * 0.05) * 200);
+        int wave = int(sin((frameCount + ( fx*fy )) * 0.05) * 150);
         // int wave = int(sin((frameCount + ( fx*fy )) * 0.05) * map(mouseY, 5, height, 0, 200)); // trigonomic function
       
         if ( this.isHovered()) { 
-          wave = int(sin((frameCount + ( fx*fy )) * 0.05) * map(mouseY, 5, 499, 0, 200)); 
+          wave = int(sin((frameCount + ( fx*fy )) * 0.05) * map(mouseY, this.posY, this.posY + this.tileH, 10, 100)); 
+          pg.fill(255, 0, 0);
+          pg.rect(0 + fx, 0 + fy, 20, 20);
```

Comment:
You shouldn't draw the rectangle here because we are inside a loop. Currently you are drawing it 32 (`Tile.fraqX * Tile.fraqY` ) times and any drawing operation costs performance. Although this is not your performance bottleneck here you should solve this differently. 

For example you could add a condition to `this.isHovered()` into the `display` function.